### DISCUSSION TITLE
First contact with Avalonia: How to get GPS location in Android?

### DISCUSSION THREAD

**Discussion by JosepBalague** [AvaloniaUI/Avalonia#10949] [General]
Hi folks,

I'm new in Avalonia. I'm porting an app from Xamarin.Forms (Android) to Avalonia. 

In past I used next package for GPS location: [https://github.com/jamesmontemagno/GeolocatorPlugin](https://github.com/jamesmontemagno/GeolocatorPlugin)

Is there something similar in Avalonia (Android version)?

Cheers

**Comment by timunie** [comment]
I think Xamarin.Essentials and MAUI.Essentials should work with Avalonia. You may want to create a service that you can inject based on the OS.

**Reply by timunie** [reply]
I checked the repo you linked rn and they also advised to switch to the Essential packages. Good luck 🤞

**Reply by workgroupengineering** [reply]
I think we should make Avalonia.Essentials.
